__docs__/components/all-variants.tsx

```diff
@@ -31,59 +34,125 @@ type Props = {
-                        <StyledTh scope="row" style={styles.cell}>
-                            <LabelLarge>{row.name}</LabelLarge>
-                        </StyledTh>
-                        {columns.map((col) => (
-                            <StyledTd
-                                key={col.name}
-                                style={[
-                                    styles.cell,
-                                    {
-                                        border: `${border.width.hairline}px dashed ${semanticColor.border.primary}`,
-                                    },
-                                ]}
-                            >
-                                {children({
-                                    ...row.props,
-                                    ...col.props,
-                                })}
-                            </StyledTd>
+        <>
+            {layout === "responsive" && (
+                <StyledTable style={[styles.table]}>
+                    <thead>
+                        <tr>
+                            <StyledTh style={styles.cell}>
+                                <LabelLarge>Category / State</LabelLarge>
+                            </StyledTh>
+                            {columns.map((col, index) => (
+                                <StyledTh
+                                    key={index}
+                                    scope="col"
+                                    style={styles.cell}
+                                >
+                                    <LabelLarge>{col.name}</LabelLarge>
+                                </StyledTh>
+                            ))}
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {rows.map((row, idx) => (
+                            <tr key={idx}>
+                                <StyledTh scope="row" style={styles.cell}>
+                                    <LabelLarge>{row.name}</LabelLarge>
+                                </StyledTh>
+                                {columns.map((col) => (
+                                    <StyledTd
+                                        key={col.name}
+                                        style={[
+                                            styles.cell,
+                                            {
+                                                border: `${border.width.hairline}px dashed ${semanticColor.border.primary}`,
+                                            },
+                                        ]}
+                                    >
+                                        {children(
+                                            {
+                                                ...row.props,
+                                                ...col.props,
+                                            },
+                                            `${row.name} ${col.name}`,
```

Comment:
Oh interesting, I just did something super similar in #2509!

Reply:
Nice!! I was also addressing a11y warnings in the new stories! I see in #2509 we add `aria-label` to the props. I ended up adding a new argument for the `name` so it could be used for a unique `aria-label`. I was thinking not all components may support an `aria-label` at the root level so providing it as an arg keeps it flexible! What do you think?